staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -183,6 +183,88 @@ func WithMaxInFlightLimit(
 	})
 }
 
+// WithMaxInFlightPerUserLimit limits the number of in-flight requests to buffer size of the passed in channel.
+func WithMaxInFlightPerUserLimit(
+	handler http.Handler,
+	limit int,
+	longRunningRequestCheck apirequest.LongRunningRequestCheck,
+) http.Handler {
+	if limit == 0 {
+		return handler
+	}
+
+	userLimits := struct {
+		lock  sync.Mutex
+		chans map[string]chan bool
+	}{
+		chans: make(map[string]chan bool),
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		requestInfo, ok := apirequest.RequestInfoFrom(ctx)
+		if !ok {
+			handleError(w, r, fmt.Errorf("no RequestInfo found in context, handler chain must be wrong"))
+			return
+		}
+
+		// Skip tracking long running events.
+		if longRunningRequestCheck != nil && longRunningRequestCheck(r, requestInfo) {
+			handler.ServeHTTP(w, r)
+			return
+		}
+
+		var username string
+		if currUser, ok := apirequest.UserFrom(ctx); ok {
+			username = currUser.GetName()
+		}
+
+		var c chan bool
+		if limit != 0 {
```

Comment:
isn't this always true?

Reply:
My fault. Fixed in 7ddf66baa95528c45ecb5ac12a080cddeff7bb4b